components/brave_ads/core/internal/ads/serving/choose/eligible_ads_predictor_util.h

```diff
@@ -159,6 +164,43 @@ CreativeAdPredictorMap<T> ComputePredictorFeaturesAndScores(
   return creative_ad_predictors_with_features;
 }
 
+template <typename T>
+std::vector<int> ComputeVoteRegistry(
+    const std::vector<T>& creative_ads,
+    const TextEmbeddingHtmlEventList& text_embedding_html_events) {
+  DCHECK(!creative_ads.empty());
+
+  std::vector<int> vote_registry(creative_ads.size(), 0);
+
+  for (const auto& text_embedding_html_event : text_embedding_html_events) {
+    std::vector<float> similarity_scores;
+
+    for (const auto& creative_ad : creative_ads) {
+      const ml::VectorData ad_embedding(creative_ad.embedding);
+      const ml::VectorData page_text_embedding(
+          text_embedding_html_event.embedding);
+      const float similarity_score =
+          ad_embedding.ComputeSimilarity(page_text_embedding);
```

Comment:
can we not call creative_ad.embedding.ComputerSimilarity` instead of assigning on line 179?

Reply:
`creative_ad.embedding` is a vector of floats, we have to convert it first to VectorData to get the vector-related functionality